executor/runtime/docker/docker.go

```diff
@@ -806,8 +739,76 @@ func prepareNetworkDriver(parentCtx context.Context, cfg Config, c *runtimeTypes
 	return nil
 }
 
+func (r *DockerRuntime) doSetupSSHdContainer(ctx context.Context, containerName *string) error { // nolint: gocyclo
+	tmpImageInfo, err := imageExists(ctx, r.client, r.cfg.ContainerSSHDImage)
+	if err != nil {
+		return err
+	}
+
+	if tmpImageInfo == nil {
+		err = pullWithRetries(ctx, r.metrics, r.client, r.cfg.ContainerSSHDImage, doDockerPull)
+		if err != nil {
+			return err
+		}
+		_, _, err = r.client.ImageInspectWithRaw(ctx, r.cfg.ContainerSSHDImage)
+		if err != nil {
+			return err
+		}
+	}
+
+	// so we replace @ with - to match Docker's image naming scheme, a la:
+	// [a-zA-Z0-9][a-zA-Z0-9_.-]
+	noDashes := strings.Replace(r.cfg.ContainerSSHDImage, ":", "-", -1)
+	noAts := strings.Replace(noDashes, "@", "-", -1)
+	noSlashes := strings.Replace(noAts, "/", "_", -1)
+	*containerName = "titus-sshd-" + noSlashes
```

Comment:
Why is this based on the image, and not on the ID of the container that it's meant to ssh into?

Reply:
because this container / volume is shared amongst all containers that refer to the same image. As you can see, it's not deleted / terminated. This way, we can reuse the container for "VolumesFrom"

Reply:
So the container is shared, but there's a separate running sshd executable per container to ssh into (as per the titus-sshd systemd unit below)?

Reply:
Yes.